cloud-prepare/vpc-peering.md

```diff
@@ -0,0 +1,210 @@
+# Submariner VPC Peering Support
+
+This document describes the technical details and impact of our proposal design
+to include Virtual Private Cloud (VPC) Peering support for Submariner and
+`subctl` located at this [Pull Request](https://github.com/submariner-io/cloud-prepare/pull/190).
+
+## Abstract
+This idea was born meanwhile the CockroachDB project focused to deploy a
+multi-cluster setup of CockroachDB and communicate the different clusters across
+regions using Submariner.
+
+One of the requirements generated from that project was to deploy CockroachDB on
+AWS in different regions at the same time, using VPC Peering and configure VXLan
+(instead of IPsec) to increase network performance because VPC Peering
+connections are already secure and encrypted.
+
+In order to achieve that, we observed neither the Operator nor the `subctl` CLI
+supported VPC Peering setup, configuration or deployment. For that reason, we
+decided to provide such functionality to cloud-prepare and subctl.  With a set
+of sub-commands in subctl, developers will automatically provision and configure
+a VPC Peering between two clusters of the same cloud provider. Once
+cloud-prepare allows VPC Peering, the Submariner Operator will include the VPC
+Peering configuration as well.
+
+## Design
+
+The following sections include the technical aspects of this feature adding and
+how it works.
+
+### New `subctl` sub-commands
+
+We purpose to add some subcommands to `subctl` to use the new VPC Peering
+features described on this document.
+
+#### VPC Peering Subcommands
+
+This will be how new added sub-commands would be like.
+
+```bash
+$ ./bin/subctl cloud VPC Peering
```

Comment:
I think we can consider making this a flag for the cloud prepare command  too, like --peer( or some other name ) network-info

Reply:
the problem with the flag approach is that it doesn't follow the same setup as the other cases because we have to specify between which clusters we want to do the peering

Reply:
Yeah, we discussed this early on. Flag can't work coz it explicitly requires at least two. Only other potion would be to setup peering between many with a single command. Which can be done in a future enhancement, with input taking list of clouds from a yaml/json file.